docs/source-app/workflows/add_web_ui/panel/panel_frontend.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import sys
+
+import panel as pn
+
+import lightning_app as lapp
+from lightning_app.utilities.imports import requires
+
+logger = logging.getLogger("PanelFrontend")
+logger.setLevel(logging.DEBUG)
+
+handler = logging.StreamHandler(sys.stdout)
+formatter = logging.Formatter("%(asctime)s - %(message)s")
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+
+
+class PanelFrontend(lapp.LightningWork):
+    @requires("panel")
+    def __init__(self, render_fn, parallel=True, **kwargs):
+        super().__init__(parallel=parallel, **kwargs)
+        self._render_fn = render_fn
+        self._server = None
+        self.requests = 0
+        logger.info("init finished")
+
+    def _fast_initial_view(self):
+        self.requests += 1
+        logger.info("Session %s started", self.requests)
+        if self.requests == 1:
+            return pn.pane.HTML("<h1>Please refresh the browser to see the app.</h1>")
+        else:
+            return self._render_fn(self)
+
+    def run(self):
+        logger.info("run start")
+        if not self._server:
+            logger.info("LitPanel Starting Server")
+            self._server = pn.serve(
+                {"/": self._fast_initial_view},
+                port=self.port,
+                address=self.host,
+                websocket_origin="*",
+                show=False,
+                # threaded=True,

```

Comment:
@philiprudiger, @tchaton . Here I need to figure out what the best way to serve Panel apps is.

- Is it `pn.serve({"/": some_function}` or `panel serve script.py`? Why?
- If `pn.serve` what value of `threaded` should I use? True/ False?
- If `threaded=True` setting an attribute like `self.counter += 1` does not seem to update the global app state. Why?